app/pages/household/[houseId]/utility/[utilityId]/index.tsx

```diff
@@ -27,47 +27,165 @@ const Login = () => {
-      {
-        "latestBooking.from": Timestamp.now(),
-        "latestBooking.to": Timestamp.now(),
-        "latestBooking.userId": currentUser?.userId,
+  const [amenityData, loading, error, snapshot] = useDocumentData(amenityRef);
+  const [validBooking, setValidity] = useState(-1);
+  const [bookingData, setBookingData] = useState({
+    bookingDesc: undefined,
+    bookingTo: undefined,
+    bookingFrom: undefined,
+  });
+  const bookingRef = collection(db, "booking");
+
+  useEffect(() => {
+    if (amenityData) {
+      // Checks if amenity is already booked
+      var utilityAvail = isUtilityAvail(
+        bookingData.bookingFrom,
+        bookingData.bookingTo
+        // amenityData
+      );
+
+      // Checks if user is booking into the past
+      var bookingMakesSense = datesWellFormed(
+        bookingData.bookingFrom,
+        bookingData.bookingTo
+      );
+      console.log("Are the dates well formed?:", bookingMakesSense);
+      console.log("Is current utility available? ", utilityAvail);
+
+      // If booking dont make sense OR its unavailable
+      if (bookingMakesSense == false || utilityAvail == false) {
+        setValidity(0);
+        console.log("BOOKING INVALID");
       }
-    );
+    }
+  }, [bookingData]); // Unsure what to do with this
+
+  const handleBooking = async (event: any) => {
+    // Prevents page from auto reloading every form submit
+    event.preventDefault();
+
+    const bookingDesc = event.target.elements.desc.value;
+    const bookingTo = event.target.elements.to.value;
+    const bookingFrom = event.target.elements.from.value;
+    setBookingData({ bookingDesc, bookingTo, bookingFrom });
+
+    if (amenityData) {
+      // Checks if amenity is already booked
+      var utilityAvail = isUtilityAvail(
+        bookingFrom,
+        bookingTo
+        // amenityData
+      );
+      // Checks if user is booking into the past
+      var bookingMakesSense = datesWellFormed(bookingFrom, bookingTo);
+
+      console.log("UTIL AVAIL?", utilityAvail);
+      console.log("BOOKING GOOD?", bookingMakesSense);
 
-    console.log("Document written with ID: ", docRef.id);
+      // Do below if ((bookingMakesSense && alreadyBooked) == true)
+      if (bookingMakesSense == true && utilityAvail == true) {
+        console.log("HELLO WORLDDDDDDD");
+        const docRef = await addDoc(bookingRef, {
+          utilityId,
+          desc: bookingDesc,
+          userId: currentUser?.userId,
+          householdId: houseId,
+          from: bookingFrom,
+          to: bookingTo,
+        });
+
+        const updateAmenity = await updateDoc(amenityRef, {
+          "latestBooking.from": bookingFrom,
+          "latestBooking.to": bookingTo,
+          "latestBooking.userId": currentUser?.userId,
+          "latestBooking.bookingID": docRef.id,
+        });
+        setValidity(1);
+
+        // ToDo:
+        // 1. Change last booking system to an array of bookings?
```

Comment:
I think it has to be an array, or query the whole list of bookings for the amenity after the current time. Since right now, you're only checking clashes against the newest booking for the amenity, which means it can still clash with other previous bookings made in the future...
Querying might be better because storing and managing the array of future booking will probably be unnecessary work plus duplicate data in the db.